src/main/scala/zio/prelude/fx/ZPure.scala

```diff
@@ -575,26 +575,26 @@ object ZPure {
     final val Modify  = 6
   }
 
-  private final case class Succeed[+A](value: A)                        extends ZPure[Any, Nothing, Any, Nothing, A] {
+  private final case class Succeed[+A](value: A)                     extends ZPure[Any, Nothing, Any, Nothing, A] {
     override def tag: Int = Tags.Succeed
   }
-  private final case class Fail[+E](error: E)                           extends ZPure[Any, Nothing, Any, E, Nothing] {
+  private final case class Fail[+E](error: E)                        extends ZPure[Any, Nothing, Any, E, Nothing] {
     override def tag: Int = Tags.Fail
   }
-  private final case class Modify[-S1, +S2, +E, +A](run: S1 => (S2, A)) extends ZPure[S1, S2, Any, E, A]             {
```

Comment:
Good catch removing the unnecessary `E` type parameter here.

Reply:
Dotty basically force me to :)